typings/styled-function.d.ts

```diff
@@ -4,7 +4,9 @@
  * @see {@link https://github.com/paypal/glamorous/blob/master/src/create-glamorous.js#L28-L131}
  */
 
-export type GlamorousComponent<P> = React.ComponentClass<P & ExtraGlamorousProps>
+export type GlamorousComponent<P> = 
+  | React.ComponentClass<P & ExtraGlamorousProps>
+  | React.StatelessComponent<P & ExtraGlamorousProps>
```

Comment:
`glamorousComponentFactory` returns a component class only. So it cannot be a union or else it would break HOC that wraps a Glamorous component. I think you meant to update https://github.com/paypal/glamorous/blob/master/typings/glamorous.d.ts#L25.